tests/rptest/clients/kafka_cli_tools.py

```diff
@@ -277,16 +272,25 @@ def reassign_partitions(self,
         else:
             raise NotImplementedError(f"Unknown operation: {operation}")
 
+        json_file = tempfile.NamedTemporaryFile(delete=False)
         output = None
+        try:
+            with open(json_file.name, "w") as outfile:
+                json.dump(reassignments, outfile)
```

Comment:
good point. i think because the new fd opened is closed that the correct flushing and visibility will be fine. but you're right we could clean this up with your suggestion so we don't have the extra open fd.